### ISSUE TITLE
Failed to read in local execute job - Missing positional argument for HttpRequest when submitting job on local compute

### ISSUE THREAD

**Issue by amrsharaf** [Azure/azure-sdk-for-python#28139]
- **Package Name**:  azure-ai-ml
- **Package Version**: 1.2.0
- **Operating System**: MacOS Ventura 13.1
- **Python Version**: 3.8.15

**Describe the bug**

I'm trying to run a simple AML job on my local compute, the job fails, with the error: Failed to read in local execute job. Upon further debugging, I found that the error was in this line:

https://github.com/Azure/azure-sdk-for-python/blob/main/sdk/ml/azure-ai-ml/azure/ai/ml/operations/_local_job_invoker.py#L149

`response = requests_pipeline.post(url=url, json=body, headers={"Authorization": "Bearer " + token})`

The fix would be using a positional rather than a keyword argument like this:

`response = requests_pipeline.post(url, json=body, headers={"Authorization": "Bearer " + token})`

**To Reproduce**

Run the following code:

```python
#import required libraries
from azure.ai.ml import MLClient, command
from azure.ai.ml.entities import Environment
from azure.identity import DefaultAzureCredential

#connect to the workspace
ml_client = MLClient.from_config(DefaultAzureCredential())

# set up pytorch environment
env = Environment(
    image='mcr.microsoft.com/azureml/openmpi3.1.2-ubuntu18.04',
    name='pytorch-env'
)

# define the command
command_job = command(
    command='echo hi',
    environment=env,
    compute='local',
)

returned_job = ml_client.jobs.create_or_update(command_job)
print(returned_job)
```

**Expected behavior**

Job on local compute should run and finish successfully. Even after fixing this bug, my local job still failed due to docker registry authentication failure, I fixed that too by using default values for username and password, but I'll open another issue / ticket for this. The job would still fail though due to an error in executing the vm bootstrapper. It would be really helpful if someone can provide a running version of the code for submitting a job on local compute, I can only fine sdk v1 examples.

**Comment by jacobdanovitch** [comment]
I'm hitting this as well.

**Comment by mazumdarabhinav** [comment]
I am facing the same issue. Can't execute command job in local

**Comment by TimbusCalin** [comment]
Same issue

**Comment by kamilpz** [comment]
Any updates on this?

**Comment by corticalstack** [comment]
Any updates on this please, as have same issue?

**Comment by vaaale** [comment]
I have the same issue. Would be really nice if this could be fixed soon.

**Comment by corticalstack** [comment]
Any update please?

**Comment by amishra-vn** [comment]
Do you happen to have any updates on this?

**Comment by diondrapeck** [comment]
Hi everyone, sorry for the lack of updates on this one. A PR was recently merged to main to address this: #31285. It will be included in the next release.
